frontend/include/chpl/resolution/resolution-types.h

```diff
@@ -1974,10 +1941,615 @@ class CallScopeInfo {
   const PoiScope* poiScope() const { return poiScope_; }
 };
 
-class ResolvedParamLoop;
+class ResolvedFunction;
+struct Resolver;
 
 /**
+  This class is used to manage mutable state that may be required while
```

Comment:
Please move ResolutionContext to its own header file frontend/include/chpl/resolution/ResolutionContext.h